osf/models/contributor.py

```diff
@@ -41,6 +44,19 @@ class Meta:
         # NOTE: Adds an _order column
         order_with_respect_to = 'node'
 
+    @cached_property
+    def has_institutional_request(self):
+        """
+        Checks if user a has requested a node using the institutional access request feature.
+        """
+        return self.user.has_institutional_request(self.node)
```

Comment:
This is still going to make the query once per contributor when loading the page. Could we make a boolean field on the `NodeRequest` (or maybe `AbstractRequest`, if we want to plan for the future) model to hold this information?